SQL/Archive/2018-05-28_publication_schema.sql

```diff
@@ -0,0 +1,159 @@
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+-- Separate table for Keywords
+DROP TABLE IF EXISTS publication_keyword;
+CREATE TABLE `publication_keyword` (
+  `PublicationKeywordID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Label` varchar(255) NOT NULL,
+  PRIMARY KEY(`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+-- Publication - Keyword relational table
+DROP TABLE IF EXISTS publication_keyword_rel;
+CREATE TABLE `publication_keyword_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationKeywordID` int(10) unsigned NOT NULL,
+  CONSTRAINT `UC_publication_keyword_1` UNIQUE (PublicationID, PublicationKeywordID),
+  CONSTRAINT `FK_publication_keyword_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_keyword_2` FOREIGN KEY(`PublicationKeywordID`) REFERENCES `publication_keyword` (`PublicationKeywordID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+
+DROP TABLE IF EXISTS publication_collaborator;
+CREATE TABLE `publication_collaborator` (
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL AUTO_INCREMENT,
+  `Name` varchar(255) NOT NULL,
+  `Email` varchar(255),
+  PRIMARY KEY(`PublicationCollaboratorID`),
+  UNIQUE (`Name`,`Email`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+DROP TABLE IF EXISTS publication_collaborator_rel;
+CREATE TABLE `publication_collaborator_rel` (
+  `PublicationID` int(10) unsigned NOT NULL,
+  `PublicationCollaboratorID` int(10) unsigned NOT NULL,
+  CONSTRAINT `UC_publication_collaborator_1` UNIQUE (PublicationID, PublicationCollaboratorID),
+  CONSTRAINT `FK_publication_collaborator_1` FOREIGN KEY(`PublicationID`) REFERENCES `publication` (`PublicationID`),
+  CONSTRAINT `FK_publication_collaborator_2` FOREIGN KEY(`PublicationCollaboratorID`) REFERENCES `publication_collaborator` (`PublicationCollaboratorID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+-- Publication - Variable of Interest  relational table
+DROP TABLE IF EXISTS publication_parameter_type_rel;
+CREATE TABLE `publication_parameter_type_rel` (
+    `PublicationID` int(10) unsigned NOT NULL,
+    `ParameterTypeID` int(10) unsigned NOT NULL,
+    CONSTRAINT `UC_publication_parameter_type_1` UNIQUE (PublicationID, ParameterTypeID),
+    CONSTRAINT `FK_publication_parameter_type_rel_1` FOREIGN KEY (`PublicationID`) REFERENCES `publication` (`PublicationID`),
+    CONSTRAINT `FK_publication_parameter_type_rel_2` FOREIGN KEY (`ParameterTypeID`) REFERENCES `parameter_type` (`ParameterTypeID`)
+) ENGINE=InnoDB DEFAULT CHARSET='utf8';
+
+DROP TABLE IF EXISTS publication_test_names_rel;
+CREATE TABLE `publication_test_names_rel` (
+    `PublicationID` int(10) unsigned NOT NULL,
+    `TestNameID` int(10) unsigned NOT NULL,
+    CONSTRAINT `FK_publication_test_names_rel_1` FOREIGN KEY (`PublicationID`) REFERENCES `publication` (`PublicationID`),
+    CONSTRAINT `FK_publication_test_names_rel_2` FOREIGN KEY (`TestNameID`) REFERENCES `test_names` (`ID`)
```

Comment:
Is test_names already exists? (currently limited access)
Should be singular `test_name`

Can a publication have mulitple test names? if yes, does the order matter?

Reply:
`test_names` is an existing table. i kept it plural for consistency
and yes, publications can have multiple test names but no the order does not matter

Reply:
Ok thanks for clarifying the situation. All good in that case.